modules/dnn/test/test_layers_1d.cpp

```diff
@@ -245,4 +245,35 @@ INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_Elemwise_1d_Test, Combine(
 /*operation*/           Values("div", "prod", "max", "min", "sum")
 ));
 
+// WORKING FOR 1D
+typedef testing::TestWithParam<tuple<int>> Layer_Permute_Test;
+TEST_P(Layer_Permute_Test, Accuracy)
+{
+    LayerParams lp;
+    lp.type = "Permute";
+    lp.name = "PermuteLayer";
+
+    int order[] = {0}; // Since it's a 0D tensor, the order remains [0]
+    lp.set("order", DictValue::arrayInt(order, 1));
+
+    int dims = get<0>(GetParam());
+
+    Ptr<PermuteLayer> layer = PermuteLayer::create(lp);
+    std::vector<int> input_shape = {dims};
+    std::vector<int> output_shape = {dims};
+
+    Mat input(dims, input_shape.data(), CV_32F);
+    cv::randn(input, 0.0, 1.0);
+    Mat output_ref(dims, output_shape.data(), CV_32F, input.data);
+
+    std::vector<Mat> inputs{input};
+    std::vector<Mat> outputs;
+
+    runLayer(layer, inputs, outputs);
+    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
+    normAssert(output_ref, outputs[0]);
+}
+INSTANTIATE_TEST_CASE_P(/*nothing*/,  Layer_Permute_Test,
+/*input blob shape*/    Values(0, 1));
```

Comment:
Better to parametrize among shape:
```cpp
typedef testing::TestWithParam<std::vector<int>> Layer_Permute_Test;
TEST_P(Layer_Permute_Test, Accuracy)
{
    LayerParams lp;
    lp.type = "Permute";
    lp.name = "PermuteLayer";

    int order[] = {0}; // Since it's a 0D tensor, the order remains [0]
    lp.set("order", DictValue::arrayInt(order, 1));

    Ptr<PermuteLayer> layer = PermuteLayer::create(lp);
    std::vector<int> input_shape = GetParam();
    std::vector<int> output_shape = input_shape;

    Mat input(input_shape.size(), input_shape.data(), CV_32F);
    cv::randn(input, 0.0, 1.0);
    Mat output_ref(output_shape.size(), output_shape.data(), CV_32F, input.data);

    std::vector<Mat> inputs{input};
    std::vector<Mat> outputs;

    runLayer(layer, inputs, outputs);
    ASSERT_EQ(shape(output_ref), shape(outputs[0]));
    normAssert(output_ref, outputs[0]);
}
INSTANTIATE_TEST_CASE_P(/*nothing*/,  Layer_Permute_Test,
/*input blob shape*/  ValuesIn(std::vector<std::vector<int>>{{}, {1}, {10}}));
```

Reply:
the purpose is to test 0/1 cases, which are scalars. 
Not sure how the parametrization with shapes should look like

Reply:
If we only test scalars.

Reply:
are you talking about 1D case of `[x1, x2, ...., xn]` form?

Reply:
Yes, but this will 1D and if you can expand this PR to cover a vector - that would be great addition.